ouroboros-consensus/src/Ouroboros/Consensus/Node.hs

```diff
@@ -90,26 +106,30 @@ fromCoreNodeId (CoreNodeId n) = CoreId n
 -------------------------------------------------------------------------------}
 
 -- | Interface against running relay node
-data NodeKernel m up blk = NodeKernel {
-      -- | Get current chain
-      getCurrentChain   :: STM m (Chain blk)
```

Comment:
Let's bring this back.

Reply:
Done, but the type is now `m (Chain blk)` instead of `STM m  (Chain blk)` as the `ChainDB` will need to read blocks from disk.

Reply:
Oh, hmm. Do we really want to support that? Read the _entire_ chain from disk? I guess I had envisioned this to be a wrapper getting the current fragment... Or maybe we should just re-expose the ChainDB...? 

Maybe I didn't think this through enough when I wrote the comment :flushed:

Reply:
Yeah, I think re-exposing the `ChainDB` is better.

Reply:
Hmm, this means we can/should remove `getExtLedgerState` as it is just calling `ChainDB.getCurrentLedger` on the `ChainDB`.

Reply:
Works for me.